### ISSUE TITLE
CREAM (Ethereum, BSC, Arbitrum, Polygon)

### ISSUE THREAD

**Issue by ishraq8** [messari/subgraphs#945]
Note: Ethereum deprecated on Oct. 17, but subgraph still picks up data.  
**Hard code subgraph not to collect data after Oct. 17 on Ethereum deployment**
Ethereum
Subgraph Reviewed | https://subgraphs.messari.io/subgraph?endpoint=https://api.thegraph.com/subgraphs/name/messari/cream-finance-ethereum&tab=protocol
-- | --
Date Reviewed | August 5th 2022
Schema Version | 2.0.1
Subgraph Version | 1.1.0
Methodology Version | 1.0.0

Protocol |   |   |  
-- | -- | -- | --
Section | Value | Status | Comments
financialsDailySnapshots | totalValueLockedUSD | NOT OK | - The TVL numbers are big difference compare to DefiLlama (95% to 97% Difference), dates after OCT 15 2021 cannot be used

Pool Overview |   |   |  
-- | -- | -- | --
All pool data | totalValueLockedUSD | not ok | TVL off on all pools

Arbitrum
Subgraph Reviewed | https://subgraphs.messari.io/subgraph?endpoint=https://api.thegraph.com/subgraphs/name/messari/cream-finance-arbitrum&tab=protocol
-- | --
Date Reviewed | August 5th 2022
Schema Version | 2.0.1
Subgraph Version | 1.1.0
Methodology Version | 1.0.0

Protocol |   |   |  
-- | -- | -- | --
Section | Value | Status | Comments
financialsDailySnapshots | dailyActiveUsers | not ok | huge spike on june 29,2022

BSC
Subgraph Reviewed | https://subgraphs.messari.io/subgraph?endpoint=https://api.thegraph.com/subgraphs/name/messari/cream-finance-bsc&tab=protocol
-- | --
Date Reviewed | August 5th 2022
Schema Version | 2.0.1
Subgraph Version | 1.1.0
Methodology Version | 1.0.0

Protocol |   |   |  
-- | -- | -- | --
Section | Value | Status | Comments
financialsDailySnapshots | totalValueLockedUSD | NOT OK | large tvl difference between front end
financialsDailySnapshots | dailySupplySideRevenueUSD | not ok | low likely due to poor tvl
financialsDailySnapshots | cumulativeSupplySideRevenueUSD | not ok | low likely due to poor tvl
financialsDailySnapshots | dailyProtocolSideRevenueUSD | NOT OK | low likely due to poor tvl
financialsDailySnapshots | cummulativeProtocolSideRevenueUSD | NOT OK | low likely due to poor tvl
financialsDailySnapshots | dailtyTotalRevenueUSD | NOT OK | low likely due to poor tvl
financialsDailySnapshots | cummulativeTotalRevenueUSD | NOT OK | low likely due to poor tvl
financialsDailySnapshots | totalBorrowUSD | NOT OK | large difference between front end, very low borrow
financialsDailySnapshots | cummulativeBorrowUSD | NOT OK | wrong due to low borrow
financialsDailySnapshots | totalDepositBalanceUSD | NOT OK | looks low due to tvl

Protocol |   |   |  
-- | -- | -- | --
Section | Value | Status | Comments
financialsDailySnapshots | totalValueLockedUSD | NOT OK | - The TVL numbers are big difference compare to DefiLlama (95% to 97% Difference), dates after OCT 15 2021 cannot be used

Polygon
Subgraph Reviewed | https://subgraphs.messari.io/subgraph?endpoint=https://api.thegraph.com/subgraphs/name/messari/cream-finance-polygon&tab=protocol
-- | --
Date Reviewed | August 5th 2022
Schema Version | 2.0.1
Subgraph Version | 1.1.0
Methodology Version | 1.0.0

Protocol |   |   |  
-- | -- | -- | --
Section | Value | Status | Comments
financialsDailySnapshots | dailyActiveUsers | not ok | Huge spike june 4, 2022

**Comment by this-username-is-taken** [comment]
another reference: https://github.com/messari/subgraphs/issues/748

**Comment by dmelotik** [comment]
@ishraq8 are these issues all post oct 17? ie, aside from the daily active user issues will the data be good if we just cutoff at oct 17?

**Comment by ishraq8** [comment]
October 17 only for ethereum. the issues raised were pre oct 17 for ethereum. The other chains should be full time.

**Comment by dmelotik** [comment]
@ishraq8 Neither of the `dailyActiveUser` spikes seem too crazy to me. Could you take a second look at those? 

Since it is not an issue on any other compound forks, i don't really think there is anything wrong with the logic.

**Comment by ishraq8** [comment]
Agreed, they are high, but not ridiculously high. This will pass Arbitrum and polygon subgraphs for prod. Any thoughts on the TVL issue on ethereum and BSC, it is weird that they are so off but other chains match perfectly with defillama. @dmelotik

**Comment by dmelotik** [comment]
I found the issue! The prices in those markets are represented in ETH and BNB respectively. So it looks good now in my tests! The only issue I am facing is getting the BNB price on chain from 9/2/2020-10/2/2020... I'm sure there is a way, but it is likely hidden under some cobwebs. Even pancakeswap v1 on bsc didn't have a real pair back then

**Comment by ishraq8** [comment]
Cream issues are resolved. Front end for BSC and Ethereum match our subgraph for TVL, Dune user metrics match our subgraph and Defi pulse matches our metrics. Defillama data is incorrect.

TLDR: Cream is done @this-username-is-taken
